src/cpp/rtps/builtin/discovery/endpoint/EDPServer2.cpp

```diff
@@ -35,9 +35,8 @@ namespace rtps {
 
 bool EDPServer2::createSEDPEndpoints()
 {
-    // Assert that PDP is a SERVER
-    PDPServer2* pPDP = dynamic_cast<PDPServer2*>(mp_PDP);
-    assert(pPDP);
+    // Assert that there is PDP SERVER
+    assert(mp_PDP);
```

Comment:
I think the idea was to assert that the `mp_PDP` is in fact pointing to a `PDPServer2*`
```suggestion
    assert(dynamic_cast<PDPServer2*>(mp_PDP));
```